stable/newrelic-infrastructure/templates/daemonset.yaml

```diff
@@ -32,21 +32,30 @@ spec:
           securityContext:
             privileged: true
           env:
+            {{- if .Values.licenseKey }}
             - name: NRIA_LICENSE_KEY
               valueFrom:
                 secretKeyRef:
                   name: {{ template "newrelic.fullname" . }}-config
                   key: license
+            {{- end }}
+            {{- if and .Values.customSecret (not .Values.licenseKey) }}
+            - name: "NRIA_LICENSE_KEY"
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Values.customSecret.name }}
```

Comment:
I'd rather see just the `name` and `key` bit bounded by the conditional. No point capturing the first four lines as well.